### PR TITLE
HDDS-2926. Intermittent failure in TestRecon due to thread timing.

### PR DESCRIPTION
**Pull request by swagle**
## What changes were proposed in this pull request?
- Removed Thread.sleep and actually made deterministic calls.
- Refactored service start/stop code to make it test friendly.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-2926

## How was this patch tested?
The integration test passed.

### CODE REVIEW

**Comment by swagle** [comment]
Thanks @avijayanhwx for suggestions.

**Comment by swagle** [comment]
UT failure is related will update patch with a fix.

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/ReconContainerDBProvider.java [hunk 1]
```diff
@@ -79,8 +81,8 @@ public DBStore get() {
     return dbStore;
   }
 
-  private static DBStore getDBStore(OzoneConfiguration configuration,
-                            ReconUtils reconUtils, String dbName) {
+  static DBStore initializeDBStore(OzoneConfiguration configuration,
```
Comment (avijayanhwx) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/ReconContainerDBProvider.java]: This can be private.

**Comment by adoroszlai** [comment]
Thanks for working on this @swagle.  The change LGTM, and the test passed locally multiple times.  But now there is some conflict in the branch.

**Comment by avijayanhwx** [comment]
LGTM +1.

**Pull request merged**